.github/workflows/pull_request.yml

```diff
@@ -37,6 +37,20 @@ jobs:
         swift build
       enable_windows_docker: false
 
+  tests_macos:
```

Comment:
Should this be part of `test_with_docker`? I would imagine that repos usually define one job that tests all the platforms. I would maybe rename `test_with_docker` to `Test` in that case and then we just have `test_without_docker` left to test Windows without Docker (and could probably rename to `test_windows_without_docker`.